internal/client/run.go

```diff
@@ -0,0 +1,74 @@
+package client
+
+import (
+	l "log"
+	"os"
+
+	"github.com/urfave/cli"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/internal/client/config"
+	"github.com/aquasecurity/trivy/internal/rpc/client/library"
+	"github.com/aquasecurity/trivy/internal/rpc/client/ospkg"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/report"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/utils"
+)
+
+func Run(cliCtx *cli.Context) error {
+	c, err := config.New(cliCtx)
+	if err != nil {
+		return err
+	}
+	return run(c)
+}
+
+func run(c config.Config) (err error) {
+	if err = log.InitLogger(c.Debug, c.Quiet); err != nil {
+		l.Fatal(err)
```

Comment:
Here and the call to `os.Exit` makes me thinking whether the run() method should return any error? It seems that it terminates with exit code so the called cannot do anything with the returned error.

Reply:
I'd prefer returning error here. And caller could decide whether to fail or not.

Reply:
Ah, this was a hack to avoid calling `log.Fatal` in `main()`. If `log.InitLogger` fails to initialize, `log` would be nil and `log.Fatal` causes panic. But I already added the handling into `main.go`
https://github.com/aquasecurity/trivy/blob/master/cmd/trivy/main.go#L20

In short, it is unnecessary now. I fixed it. Thanks.